oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/record-requests/utils/hooks.js

```diff
@@ -65,13 +92,80 @@ export const useConfirmDialog = (formik, sendRequest, isEventModal) => {
         confirmButton: <Button positive>{i18next.t("Create and submit")}</Button>,
         onConfirm: () => {
           setConfirmDialogProps(props => ({ ...props, open: false }));
-          sendRequest(REQUEST_TYPE.CREATE, createAndSubmit);
+          onConfirm();
         }
       }
     }
 
     setConfirmDialogProps(props => ({ ...props, ...newConfirmDialogProps }));
-  };
+  }, [setSubmitting, isEventModal]);
 
   return { confirmDialogProps, confirmAction };
 }
+
+export const useRequestsApi = () => {
+  const {
+    values: formValues,
+    resetForm,
+    setSubmitting,
+    setErrors,
+  } = useFormikContext();
+
+  const setError = error => { setErrors({ api: error }); };
+
+  const callApi = async (url, method, data = formValues, doNotHandleResolve = false) => {
+    if (_isEmpty(url)) {
+      const err = new Error(i18next.t("Cannot send request. Please try again later."));
+      setError(err);
+      setSubmitting(false);
+      throw err;
+    }
+
+    const request = axios({
+      method: method,
+      url: url,
+      data: data,
+      headers: { 'Content-Type': 'application/json' }
+    });
+
+    if (doNotHandleResolve) {
+      return request;
+    }
+
+    return request
+      .then(() => {
+        resetForm();
+      })
+      .catch(error => {
+        setError(error);
+        throw error;
+      });
+  };
+
+  const createAndSubmitRequest = async (createRequestLink) => {
+    setSubmitting(true);
+    setErrors({});
+    try {
+      const createdRequest = await callApi(createRequestLink, 'post', formValues, true);
+      await callApi(createdRequest.data?.links?.actions?.submit, 'post', {}, true);
+      resetForm();
+    } catch (error) {
+      setError(error);
+      throw error;
+    }
+  };
+
+  const sendRequest = async (actionUrl, requestType) => {
```

Comment:
replace this with more user-friendly methods, handling specific actions, either `acceptRequest(event)` or `doAction(REQUEST_TYPE)`. Methods should find out necessary callApi params from request type and request.links by itself

Reply:
Instead of doing this in my components using this client:
```jsx
    const { sendRequest } = useRequestsApi();  
  const acceptActionConfirmationHandler = () => confirmAction(() => onSubmit(() => sendRequest(request.links.actions?.accept, REQUEST_TYPE.ACCEPT)), REQUEST_TYPE.ACCEPT);
  //...
  return (request.links.actions?.accept && <Button title={i18next.t("Accept request")} onClick={acceptActionConfirmationHandler} positive icon labelPosition="left" floated="right" />
```

I'd like to be able to do:

```jsx
const { acceptRequest } = useRequestsApi(request);
return <Button title={i18next.t("Accept request")} onClick={acceptRequest} positive icon labelPosition="left" floated="right" />
```

Or slightly more generic:

```jsx
const { doAction } = useRequestsApi(request);
return <Button title={i18next.t("Accept request")} onClick={() => doAction(REQUEST_TYPE.ACCEPT)} positive icon labelPosition="left" floated="right" />
```